### ISSUE TITLE
DataSource throws an error when having nested property with null parent object

### ISSUE THREAD

**Issue by TsvetinaIvanova** [telerik/kendo-ui-core#2981]
### Bug report
[Dojo](http://dojo.telerik.com/@tsveti/igiFo)

Open the console to see the error:
```
Uncaught TypeError: Cannot read property 'Name' of null
    at Object.eval [as SupplierName] (eval at getter (kendo.all.js:2048), <anonymous>:3:18)
    at y (kendo.all.js:5676)
    at init.data (kendo.all.js:5707)
    at init._readData (kendo.all.js:6468)
    at init.success (kendo.all.js:6649)
    at success (kendo.all.js:6610)
    at Object.n.success (kendo.all.js:5599)
    at i (VM1105 jquery.min.js:2)
    at Object.fireWith [as resolveWith] (VM1105 jquery.min.js:2)
    at y (VM1105 jquery.min.js:4)
```

The DataSource currently tries to access nested properties directly without checking for their existence.

### Expected/desired behavior
The DataSource should check for existence of the parent property before getting the nested value.

### Environment

* **Kendo UI version:** any

Current work around is to process the data in advance to either flatten it, or provide a default value manually for the records with null properties:
http://dojo.telerik.com/@tsveti/IBIxa/2

**Comment by avrahamcool** [comment]
in my honset opinion,
this bug is very easy to fix, and its a contained code, so it cannot harm any other code.
I dont see any reason to not fix it in the next version of kendo.

**Comment by RomainLanz** [comment]
Hey everyone!

Any news about this?

**Comment by jonhudson12** [comment]
I have found this to be a problem in the Kendo controls as well. I have found articles about using client templates to adjust output strings when encountering nulls as shown in this example:

 columns.Bound(item => item.Patient.FullName).Width(100).ClientTemplate(" #= (Patient != null ? Patient.FullName : '') # ").Title("Patient");

Grid read always fails when Patient object is null. This forces you to have to flatten your model down into a single object with no nested properties. Very cumbersome. The grid should be able to detect nulls somehow, but the syntax for lamba gets in the way, and prevents usage of null coalesce operators, so you just cant have any sub properties that could ever have a null parent.
